app/models/recipe.rb

```diff
@@ -1,20 +1,9 @@
 # This class describes a single recipe and its characteristics. 
-
-class Recipe
-
-  attr_reader :id, :name, :description, :ingredients, :time, :directions, :image_file
-
-  def initialize(args)
-    @id           = args[:id]
-    @name         = args[:name]
-    @description  = args[:description]
-    @ingredients  = args[:ingredients]
-    @time         = args[:time]
-    @directions   = args[:directions]
-    @image_file   = args[:image_file]
-  end
-
-  def to_s
-    "ID: #{@id}, Name: #{@name}, Description: #{@description}, Ingredients: #{@ingredients}, Time: #{@time}, Directions: #{@directions}."
-  end
+class Recipe < ActiveRecord::Base
+  validates :name, presence: true, length: { minimum: 3 }
+  validates :description, presence: true, length: { minimum: 3 }
+  validates :ingredients, presence: true, length: { minimum: 3 }
+  validates :time, presence: true 
+  validates :directions, presence: true, length: { minimum: 3 } 
+  validates :image_file, presence: true
```

Comment:
Need to limit all string datatypes to 254 characters. Most databases distinguish string and text by the size.